http-service/src/main/java/net/runelite/http/service/hiscore/HiscoreService.java

```diff
@@ -67,53 +102,74 @@ private HiscoreResultBuilder lookupUsername(String username) throws IOException
 			.build();
 
 		Response okresponse = RuneliteAPI.CLIENT.newCall(okrequest).execute();
-		String responseStr;
 
-		try (ResponseBody body = okresponse.body())
+		if (!okresponse.isSuccessful())
 		{
-			responseStr = body.string();
+			switch (HttpStatus.valueOf(okresponse.code()))
+			{
+				case NOT_FOUND:
+					throw new NotFoundException();
```

Comment:
This exception was only made for the cache stuff so it is under the cache package. If you want to use it here (which seems fine) move it up to a higher package.